config/crd/bases/tests.testkube.io_tests.yaml

```diff
@@ -0,0 +1,80 @@
+
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.4.1
+  creationTimestamp: null
+  name: tests.tests.testkube.io
+spec:
+  group: tests.testkube.io
+  names:
+    kind: Test
+    listKind: TestList
+    plural: tests
+    singular: test
+  scope: Namespaced
+  versions:
+  - name: v1
+    schema:
+      openAPIV3Schema:
+        description: Test is the Schema for the tests API
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: TestSpec defines the desired state of Test
+            properties:
+              scripts:
+                description: Steps is list of scripts which will be sequentially orchestrated
+                items:
+                  description: TestStepSpec will of particular type will have config
+                    for possible step types
+                  properties:
```

Comment:
perhaps we could model this using a "TestStep" base type with a descriminator property - and then derive types from that for each TestStep (ScriptTestStep, DelayTestStep, etc)  using inheritance in json schema!? the array would simply have a list of TestStep objects - see https://swagger.io/docs/specification/data-models/inheritance-and-polymorphism/